# HG changeset patch
# Parent 6821c4ce948c021cb6a16a3b3ffe9e32cc1e1d06
-Add: A 'Toggle text direction' entry in the About/Help menu if the 'gui.developer' setting is set

diff --git a/src/lang/english.txt b/src/lang/english.txt
--- a/src/lang/english.txt
+++ b/src/lang/english.txt
@@ -467,6 +467,7 @@ STR_ABOUT_MENU_ABOUT_OPENTTD            
 STR_ABOUT_MENU_SPRITE_ALIGNER                                   :Sprite aligner
 STR_ABOUT_MENU_TOGGLE_BOUNDING_BOXES                            :Toggle bounding boxes
 STR_ABOUT_MENU_TOGGLE_DIRTY_BLOCKS                              :Toggle colouring of dirty blocks
+STR_ABOUT_MENU_TOGGLE_TEXT_DIRECTION                            :Toggle text direction
 ############ about menu entries ends here
 
 ############ range for days starts (also used for the place in the highscore window)
diff --git a/src/toolbar_gui.cpp b/src/toolbar_gui.cpp
--- a/src/toolbar_gui.cpp
+++ b/src/toolbar_gui.cpp
@@ -1031,6 +1031,7 @@ enum HelpMenuEntries {
 	HME_SPRITE_ALIGNER,
 	HME_TOGGLE_BOUNDING_BOXES,
 	HME_TOGGLE_DIRTY_BLOCKS,
+	HME_TOGGLE_TEXT_DIRECTION,
 };
 
 static CallBackFunction ToolbarHelpClick(Window *w)
@@ -1050,6 +1051,9 @@ static CallBackFunction ToolbarHelpClick
 		list->push_back(new DropDownListStringItem(STR_ABOUT_MENU_TOGGLE_BOUNDING_BOXES, HME_TOGGLE_BOUNDING_BOXES, false));
 		list->push_back(new DropDownListStringItem(STR_ABOUT_MENU_TOGGLE_DIRTY_BLOCKS,   HME_TOGGLE_DIRTY_BLOCKS, false));
 	}
+	if (_settings_client.gui.developer) {
+		list->push_back(new DropDownListStringItem(STR_ABOUT_MENU_TOGGLE_TEXT_DIRECTION, HME_TOGGLE_TEXT_DIRECTION, false));
+	}
 
 	ShowDropDownList(w, list, 0, WID_TN_HELP, 140, true, true);
 	return CBF_NONE;
@@ -1161,6 +1165,7 @@ static CallBackFunction MenuClickHelp(in
 		case HME_SPRITE_ALIGNER:          ShowSpriteAlignerWindow();                     break;
 		case HME_TOGGLE_BOUNDING_BOXES:   ToggleBoundingBoxes();                         break;
 		case HME_TOGGLE_DIRTY_BLOCKS:     ToggleDirtyBlocks();                           break;
+		case HME_TOGGLE_TEXT_DIRECTION:   ToggleTextDirection();                         break;
 	}
 	return CBF_NONE;
 }
